### PR TITLE
Feature/cax 1 mtpdc@74 terraform partitions

### PR DESCRIPTION
**Pull request by algattik**
Deploy DEV environment on two "data space partitions", i.e. identical PRS deployments. Currently both deployments have the same test data loaded, and system tests run independently on each. In a later PR,  we will only load partial data to each partition, and run system test in a separate job.

The deployment configuration is in a new file dataspace-partitions.json, and the pipeline dynamically runs multiple jobs to deploy to each partition in parallel.

The previously separate _smoke test_ and  _test data load_ pipelines have been integrated into the main deployment pipeline, to reuse the matrix deployment.

Sample deployment: https://github.com/catenax/tractusx/actions/runs/1400059616

### CODE REVIEW

File: .github/workflows/partsrelationshipservice-deploy.yml [hunk 1]
```diff
@@ -54,8 +98,31 @@ jobs:
           export ARM_SUBSCRIPTION_ID=$(az account show --query "id" --output tsv)
 
           pushd terraform
-            terraform init -backend-config=key=${TERRAFORM_STATE_KEY}
-            terraform apply --var-file "../cd/${TERRAFORM_DEPLOY_VAR_FILE}" -auto-approve
+            terraform init -backend-config=key=mtpdc.$ENVIRONMENT.${{ matrix.partition.key }}.tfstate
```
Comment (marcgs) [inline .github/workflows/partsrelationshipservice-deploy.yml:101]: There are a few old `tfstate` files in the container. We could clean these up after this gets merged.

Comment (marcgs) [reply .github/workflows/partsrelationshipservice-deploy.yml:101]: We would need to manually create separate tfstate files for each partition, correct?

Comment (algattik) [reply .github/workflows/partsrelationshipservice-deploy.yml:101]: No, terraform creates them as needed

**Review by marcgs — APPROVED** [review]

File: .github/workflows/partsrelationshipservice-deploy.yml [hunk 2]
```diff
@@ -16,33 +16,77 @@ on:
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: coreservices/partsrelationshipservice
+    steps:
+      - uses: actions/checkout@v2
+      - uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: "Build and push MTPDC images to ACR"
```
Comment (ouphi) [inline .github/workflows/partsrelationshipservice-deploy.yml:60]: What is MTPDC?

File: coreservices/partsrelationshipservice/terraform/variables.tf [hunk 3]
```diff
@@ -5,7 +5,7 @@
 variable "prefix" {
   type        = string
   description = "First part of name prefix used in naming resources. Use only lowercase letters and numbers."
-  default     = "catenaprs1"
+  default     = "cxmtpdc1"
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/terraform/variables.tf:8]: what does this prefix stand for?

File: .github/workflows/partsrelationshipservice-deploy.yml (hunk 2 above)
Comment (ouphi) [reply .github/workflows/partsrelationshipservice-deploy.yml:60]: multi tier parts data chain

**Review by Izzzu — APPROVED** [review]

**Pull request merged**